tools/runners/run-mongo.js

```diff
@@ -607,38 +607,43 @@ var launchMongo = function (options) {
         });
       }
 
-      var initiateResult = yieldingMethod(
-        db.admin(), 'command', {replSetInitiate: configuration});
+      try{
+        var initiateResult = yieldingMethod(
+          db.admin(), 'command', {replSetInitiate: configuration});
+      } catch (e){
+        if(e.message !== 'already initialized') {
+          throw Error("rs.initiate error: " + e.message);
+        }
+      }
+
       if (stopped) {
         return;
       }
-      // why this isn't in the error is unclear.
-      if (initiateResult && initiateResult.documents
-          && initiateResult.documents[0]
-          && initiateResult.documents[0].errmsg) {
-        var err = initiateResult.documents[0].errmsg;
-        if (err !== "already initialized") {
-          throw Error("rs.initiate error: " +
-                      initiateResult.documents[0].errmsg);
-        }
-      }
+
       // XXX timeout eventually?
       while (!stopped) {
-        var status = yieldingMethod(db.admin(), 'command',
-                                    {replSetGetStatus: 1});
-        if (!(status && status.documents && status.documents[0])) {
-          throw status;
+        var status;
+        try{
+          status = yieldingMethod(db.admin(), 'command',
+                                      {replSetGetStatus: 1});
+        } catch (e) {
+          // Some of the expected results, come as errors.
+          status = e;
```

Comment:
I don't really like this pattern (as a reader of the code I might not expect `status` to be an error).

I'd rather the code looked like:

```js
try {
  status = ...
} catch (e) {
   if (e.myState === 6) {
     // handle non-fatal error
   } else {
     throw e;
   }
}

// now we know status is a status record.